core/src/main/java/org/apache/brooklyn/core/effector/http/HttpCommandEffector.java

```diff
@@ -181,7 +204,19 @@ private HttpRequest buildHttpRequest(String httpVerb, URI uri, Map<String, Strin
                 if (contentType == null || contentType.equalsIgnoreCase(APPLICATION_JSON)) {
                     LOG.warn("Content-Type not specified. Using {}, as default (continuing)", APPLICATION_JSON);
                     body = toJsonString(payload);
-                } else if (!contentType.equalsIgnoreCase(APPLICATION_JSON)) {
+                } else if (contentType.equalsIgnoreCase(APPLICATION_X_WWW_FORM_URLENCODE)) {
+                    if (payload instanceof Map) {
+                        for (Map.Entry<String, String> entry : ((Map<String, String>) payload).entrySet()) {
+                            try {
+                                if (!body.equals("")) body += "&";
+                                body += URLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8.toString()) + "=" + URLEncoder.encode(entry.getValue(), StandardCharsets.UTF_8.toString());
```

Comment:
Thanks @geomacy for helping me here. To be honest I don't know exactly what to do, probably best to implement the less surprising behavior. I'll try to clarify with more tests and docs